src/executor/hash_join_executor.cpp

```diff
@@ -100,7 +101,17 @@ bool HashJoinExecutor::DExecute() {
 
     // Get the hash table from the hash executor
     auto &hash_table = hash_executor_->GetHashTable();
-    auto &hashed_col_ids = hash_executor_->GetHashKeyIds();
+//    auto &hashed_col_ids = hash_executor_->GetHashKeyIds();
+    std::vector<const expression::AbstractExpression *> left_hashed_cols;
+    this->GetPlanNode<planner::HashJoinPlan>().GetLeftHashKeys(left_hashed_cols);
+
+    std::vector<oid_t> left_hashed_col_ids;
+    for (auto &hashkey : left_hashed_cols) {
+      PL_ASSERT(hashkey->GetExpressionType() == ExpressionType::VALUE_TUPLE);
+      auto tuple_value =
+          reinterpret_cast<const expression::TupleValueExpression *>(hashkey);
+      left_hashed_col_ids.push_back(tuple_value->GetColumnId());
```

Comment:
Instead of doing a push back in every iteration of the loop, can we declare the vector size from before and add the values to it using the [] operator. There are some performance degradations to doing a push_back() everytime in the loop.